src/main/java/com/dao/homework/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,57 @@
+package com.dao.homework.dao;
+
+import com.dao.homework.db.Storage;
+import com.dao.homework.lib.Dao;
+import com.dao.homework.model.Manufacturer;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        Storage.addManufacturer(manufacturer);
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> getById(Long manufacturerId) {
+        return getAllManufacturers().stream()
+                .filter(m -> Objects.equals(m.getId(), manufacturerId))
+                .findFirst();
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        if (!getAllManufacturers().contains(manufacturer)) {
+            throw new RuntimeException("Can`t find this element!");
+        }
+        manufacturer.setId(manufacturer.getId());
+        manufacturer.setCountry(manufacturer.getCountry());
```

Comment:
this code isn't flexible. We should change the logic of this method if the manufacturer object is changed
let's use the method `set` from the list for replacing an object